src/main/java/players/mcts/MCTSParams.java

```diff
@@ -1,99 +1,58 @@
 package players.mcts;
 
-import core.AbstractParameters;
-import core.interfaces.ITunableParameters;
+import core.*;
 import players.PlayerParameters;
+import players.simple.RandomPlayer;
+import java.util.*;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
+import static players.mcts.MCTSEnums.strategies.RANDOM;
 
-public class MCTSParams extends PlayerParameters implements ITunableParameters {
+public class MCTSParams extends PlayerParameters {
 
     public double K = Math.sqrt(2);
     public int rolloutLength = 10;
     public boolean rolloutsEnabled = false;
     public double epsilon = 1e-6;
+    public MCTSEnums.strategies rolloutType = RANDOM;           ;
 
+    public MCTSParams() {
+        this(System.currentTimeMillis());
+    }
     public MCTSParams(long seed) {
         super(seed);
+        addTunableParameter("K", Math.sqrt(2), Arrays.asList(0.0, 0.1, 1.0, Math.sqrt(2), 3.0, 10.0));
+        addTunableParameter("rolloutLength", 10, Arrays.asList(6, 8, 10, 12, 20));
+        addTunableParameter("rolloutsEnabled", false, Arrays.asList(false, true));
+        addTunableParameter("epsilon", 1e-6);
```

Comment:
It seems a bit repetitive adding these (and misleading since the method is called addTunableParameter, but this is not tunable). I wonder if there's an easy way to get around it... (a note for the future)

Reply:
I played with a few options here. Initially I tried reducing boilerplate by not having these as local properties, but that meant any use of a parameter had to call params.getParameterValue(name)...which was horrendous, and self-defeating.

Two specific points:
1) Parameters with a default value but no list of options **are** still tunable, so I don't think the method is ill-named. A custom-JSON file can still be used to specify a range of epsilon if you really wanted to.

2) This is still a significant reduction in boilerplate code compared to the previous version, in which default values were manually set up in code at one point, and possible values set up at another. I'm not saying this is perfect - but it feels more natural to me to set up all the parameter details in a single place in the constructor (plus, I admit, add them to the _reset() method if you want consumers to access them via a local public property).

(Open to ideas though!)

Reply:
Yeah those are good points. Not sure what would be better, maybe we'll find something else later.